spec/controllers/course/material/folders_controller_spec.rb

```diff
@@ -7,26 +7,65 @@
   with_tenant(:instance) do
     let(:user) { create(:administrator) }
     let(:course) { create(:course) }
-    let(:folder_stub) do
-      stub = create(:folder, course: course, parent: create(:folder, course: course))
-      allow(stub).to receive(:destroy).and_return(false)
-      allow(stub).to receive(:update).and_return(false)
-      allow(stub).to receive(:save).and_return(false)
-      stub
-    end
+    let(:folder_stub) { create(:folder, course: course, parent: create(:folder, course: course)) }
```

Comment:
I personally think having this kind of nested function (`create` inside another `create`) not clean. If I were you, I will have the creation of one folder first (suppose the name is `first_folder`) then create another folder with `first_folder` as the parent.

Reply:
This was already in the previous code before my PR  
`stub = create(:folder, course: course, parent: create(:folder, course: course))`  
In this case, the original developer likely feels that, since we are not making use of the `parent_folder`, there is no need to create a reference to it, and I feel the same way.

Reply:
got it